invisible_cities/io/dst_io.py

```diff
@@ -18,13 +18,15 @@ def _decode_str_columns(df):
     df        = df.apply(to_string)
     return df
 
-def load_dst(filename, group, node):
+def load_dst(filename, group, node, evt_list=None):
     """load a kdst if filename, group and node correctly found"""
     try:
         with tb.open_file(filename) as h5in:
             try:
-                table = getattr(getattr(h5in.root, group), node).read()
-                return _decode_str_columns(pd.DataFrame.from_records(table))
+                table  = getattr(getattr(h5in.root, group), node)
+                values = (table.read() if evt_list is None
+                          else [ x for evt in evt_list for x in table.read_where(f"""(event == {evt})""") ])
```

Comment:
Im not sure this is the best approach, guess this depends heavily on the user case. If the event list contains almost all the events from the file, this will be much slower than simply reading the table and selecting the desired events for two reasons:
1. lists are less efficient than numpy array
2. loop over `evt_list`. This is especially relevant in `load_dsts`where you will loop over `evt_list` that contains set of events from all the files you intend to read.
Im not sure whats the best option, maybe to read the event column, select indices and use `read_coordinates` table method? Since this PR is optimizing a piece of code guess the best would be to actually profile both time and memory of different options for your user case?

Reply:
The problem for me is not so much speed as memory, once you get to hdst and above loading several files becomes a nightmare. Specially in the case of deconvolution, where things become dramatically huge and, in my computer, is mostly impossible to load all the deconvolved events of even a single file and I must load event by event (getting the event list from the Run table). I have also found it to be pretty useful when looking into high energy peaks starting from penthesilea.  In any case, with the current implementation, the user may still use the old approach, no need to go through the event list (which, of course, for kdsts will be much slower, yeah). 

The `read_coordinates` function could do the trick, yeah, I'll test it and report on the results.

Reply:
Did some quick test, just checking the time, and the improvement on velocity with the read_coordinates approach is amazing, starting from a single file with 10 million rows dst and keeping only 250k rows after event filtering:
- load all events and selecting afterwards ~ 3 seconds, memory comsumption up to 2 gb.
- load iterating through lists ~ 41 seconds, memory comsumption up to 250 mb
- load iterating through read_coordinates ~ 3.5 seconds, memory comsumption up to 250 mb

so I would be very much inclined to use the `read_coordinates` approach and go ahead as we still keep the 'classical' functionality of the function.

Reply:
Ok, that sounds good! You are still reading the full event column to select the indices right? Does that fit in the memory?

Reply:
I also tried to see what would happen with several files, using `load_dsts`, and keeping now 250 events. Thing just explodes when trying to load several files (around 200 million rows in total). Using the coordinate thing, things are kept at around 500 mb and loads in around 1 minute, is not so fast but it manages to keep things really light.

Indeed I need to load the full event list but that does not seem to be problematic, the code would be as follows:

```
def load_dst(filename, group, node, evt_list=None):
    """load a kdst if filename, group and node correctly found"""
    try:
        with tb.open_file(filename) as h5in:
            try:
                table  = getattr(getattr(h5in.root, group), node)
                if evt_list is None:
                    values = table.read()
                else:
                    events = table.read(field='event')
                    values = table.read_coordinates(np.where(np.isin(events, evt_list)))
                return _decode_str_columns(pd.DataFrame.from_records(values, columns=table.colnames))
            except NoSuchNodeError:
                warnings.warn(f' not of kdst type: file= {filename} ', UserWarning)
    except HDF5ExtError:
        warnings.warn(f' corrupted: file = {filename} ', UserWarning)
    except IOError:
        warnings.warn(f' does not exist: file = {filename} ', UserWarning)
```